src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1632,6 +1641,7 @@ public void onFailure(Throwable e)
     @Override
     public void startBootstrap()
     {
+        Preconditions.checkState(!isCleanupRunning(), "Cleanup is running. Refusing to start bootstrap.");
```

Comment:
Should we move this to sidecar? We'll hit the NTE after we signal to start bootstrap anyway